### PR TITLE
Standardizing naming of diagnostic settings + API version update for locks and RBAC

### PR DESCRIPTION
**Pull request by rahalan**
# Change

- naming and description of diagnostic settings standardized (all start with diagnostic now)
- RBAC API version to newest updated - WARNING this one is not recognized by IntelliSense, but it is working
- lock API version updated
- API version for API management updated

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.AnalysisServices/servers/.parameters/parameters.json [hunk 1]
```diff
@@ -24,13 +24,13 @@
         "diagnosticStorageAccountId": {
             "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Storage/storageAccounts/adpsxxazsaweux001"
         },
-        "workspaceId": {
+        "diagnosticWorkspaceId": {
```
[the remark this answers is not in the document]
Comment (rahalan) [reply arm/Microsoft.AnalysisServices/servers/.parameters/parameters.json:27]: True, but this LAW has a special purpose. We even have modules (not much) that have two LAWs, one for diagnostics and one for something else. We need to make clear, what they are used for. Plus it is not just LAW, for diagnostic storage account we already use the prefix diagnostic. There are three syncs for diagnostics and they should all look the same, so the usage is clear. Even if they are not the same name as the reference dictates.

[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.AnalysisServices/servers/.parameters/parameters.json:27]: Fair point

File: arm/Microsoft.Authorization/roleAssignments/readme.md [hunk 2]
```diff
@@ -14,12 +14,12 @@ This module deploys Role Assignments.
 | :-- | :-- | :-- | :-- | :-- |
 | `condition` | string |  |  | Optional. The conditions on the role assignment. This limits the resources it can be assigned to |
 | `conditionVersion` | string | `2.0` | `[2.0]` | Optional. Version of the condition. Currently accepted value is "2.0" |
-| `delegatedManagedIdentityResourceId` | string |  |  | Optional. ID of the delegated managed identity resource |
+| `delegatedManagedIdentityResourceId` | string |  |  | Optional. Id of the delegated managed identity resource |
```
Comment (MrMCake) [inline arm/Microsoft.Authorization/roleAssignments/readme.md:17]:
```suggestion
| `delegatedManagedIdentityResourceId` | string |  |  | Optional. ID of the delegated managed identity resource |
```

File: arm/Microsoft.Compute/virtualMachineScaleSets/extensions/readme.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ This module deploys a virtual machine scale set extension.
 | :-- | :-- | :-- |
 | `extensionName` | string | The name of the extension |
 | `extensionResourceGroup` | string | The name of the Resource Group the extension was created in. |
-| `extensionResourceId` | string | The resource ID of the extension |
+| `extensionResourceId` | string | The ResourceId of the extension |
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachineScaleSets/extensions/readme.md:34]:
```suggestion
| `extensionResourceId` | string | The Resource ID of the extension |
```

File: arm/Microsoft.AnalysisServices/servers/.parameters/parameters.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (rahalan) [reply arm/Microsoft.AnalysisServices/servers/.parameters/parameters.json:27]: Can this one be resolved, or should we discuss it?

File: constructs/Microsoft.Management/managementGroup-structure/readme.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ This module has some known **limitations**:
 
 |Resource Type|ApiVersion|
 |:--|:--|
-|`Microsoft.Management/managementGroups`|2020-05-01|
+|`Microsoft.Management/managementGroups`|2017-04-01|
```
Comment (MrMCake) [inline constructs/Microsoft.Management/managementGroup-structure/readme.md]: Intended?

File: arm/Microsoft.Network/firewallPolicies/ruleGroups/readme.md [hunk 5]
```diff
@@ -43,4 +43,4 @@ For remaining properties, see [FirewallPolicyRule objects](https://docs.microsof
 
 ## Template references
 
-- [Firewallpolicies/Rulegroups](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/2020-04-01/firewallPolicies/ruleGroups)
+- ['firewallPolicies/ruleGroups' Parent Documentation](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/firewallPolicies)
```
Comment (MrMCake) [inline arm/Microsoft.Network/firewallPolicies/ruleGroups/readme.md:46]: Where is this coming from?

File: arm/Microsoft.ContainerService/managedClusters/deploy.bicep [hunk 6]
```diff
@@ -279,9 +282,9 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2021-07-01'
         enabled: httpApplicationRoutingEnabled
       }
       omsagent: {
-        enabled: (omsAgentEnabled && (!empty(workspaceId)))
+        enabled: (omsAgentEnabled && (!empty(monitoringWorkspaceId)))
         config: {
-          logAnalyticsWorkspaceResourceID: ((!empty(workspaceId)) ? workspaceId : null)
+          logAnalyticsWorkspaceResourceID: ((!empty(monitoringWorkspaceId)) ? monitoringWorkspaceId : null)
```
Comment (MrMCake) [inline arm/Microsoft.ContainerService/managedClusters/deploy.bicep:287]:
```suggestion
          logAnalyticsWorkspaceResourceID: !empty(monitoringWorkspaceId) ? monitoringWorkspaceId : null
```
Unrelated but not-necessary

File: constructs/Microsoft.Management/managementGroup-structure/readme.md (hunk 4 above)
Comment (rahalan) [reply constructs/Microsoft.Management/managementGroup-structure/readme.md:14]: No. Will fix it

File: arm/Microsoft.Network/firewallPolicies/ruleGroups/readme.md (hunk 5 above)
Comment (rahalan) [reply arm/Microsoft.Network/firewallPolicies/ruleGroups/readme.md:46]: from the readme generator

File: arm/Microsoft.EventHub/namespaces/.bicep/nested_privateEndpoint.bicep [hunk 7]
```diff
@@ -36,7 +36,7 @@ resource privateEndpoint 'Microsoft.Network/privateEndpoints@2021-03-01' = {
   }
 }
 
-resource privateDnsZoneGroups 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2020-05-01' = if (!empty(privateEndpoint_var.privateDnsZoneResourceIds)) {
+resource privateDnsZoneGroups 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2017-04-01' = if (!empty(privateEndpoint_var.privateDnsZoneResourceIds)) {
```
Comment (MrMCake) [inline arm/Microsoft.EventHub/namespaces/.bicep/nested_privateEndpoint.bicep:39]: Intended?

File: constructs/Microsoft.Management/managementGroup-structure/deploy.json [hunk 8]
```diff
@@ -191,7 +191,7 @@
         {
             "type": "Microsoft.Management/managementGroups",
             "comments": "Fake deployment, used to specify a non-existent dependency. Never deployed",
-            "apiVersion": "2020-05-01",
+            "apiVersion": "2021-04-01",
```
Comment (MrMCake) [inline constructs/Microsoft.Management/managementGroup-structure/deploy.json]: I guess that was unintended too

File: constructs/Microsoft.Management/managementGroup-structure/deploy.json [hunk 9]
```diff
@@ -238,7 +238,7 @@
             ],
 
             "type": "Microsoft.Management/managementGroups",
-            "apiVersion": "2020-05-01",
+            "apiVersion": "2021-04-01",
```
Comment (MrMCake) [inline constructs/Microsoft.Management/managementGroup-structure/deploy.json]: I guess that was unintended too

File: constructs/Microsoft.Management/managementGroup-structure/readme.md [hunk 10]
```diff
@@ -11,7 +11,7 @@ This module has some known **limitations**:
 
 |Resource Type|ApiVersion|
 |:--|:--|
-|`Microsoft.Management/managementGroups`|2020-05-01|
+|`Microsoft.Management/managementGroups`|2021-04-01|
```
Comment (MrMCake) [inline constructs/Microsoft.Management/managementGroup-structure/readme.md]: I guess that was unintended too

File: constructs/Microsoft.Management/managementGroup-structure/readme.md [hunk 11]
```diff
@@ -124,4 +124,4 @@ This template is using a **Tenant level deployment**, meaning the user/principal
 ## Additional resources
 
 - [Management group](https://docs.microsoft.com/en-us/azure/governance/management-groups/)
-- [Template reference](https://docs.microsoft.com/en-us/azure/templates/microsoft.management/managementgroups)
+- [Managementgroups](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Management/2021-04-01/managementGroups)
```
Comment (MrMCake) [inline constructs/Microsoft.Management/managementGroup-structure/readme.md]: I guess that was unintended too

File: arm/Microsoft.EventHub/namespaces/.bicep/nested_privateEndpoint.bicep (hunk 7 above)
Comment (rahalan) [reply arm/Microsoft.EventHub/namespaces/.bicep/nested_privateEndpoint.bicep]: No. Will fix it

File: constructs/Microsoft.Management/managementGroup-structure/deploy.json (hunk 9 above)
Comment (rahalan) [reply constructs/Microsoft.Management/managementGroup-structure/deploy.json:241]: No. That is correct

File: constructs/Microsoft.Management/managementGroup-structure/readme.md (hunk 10 above)
Comment (rahalan) [reply constructs/Microsoft.Management/managementGroup-structure/readme.md:14]: No. That is correct

File: constructs/Microsoft.Management/managementGroup-structure/readme.md (hunk 11 above)
Comment (rahalan) [reply constructs/Microsoft.Management/managementGroup-structure/readme.md:127]: No. That is correct

File: constructs/Microsoft.Management/managementGroup-structure/deploy.json (hunk 8 above)
Comment (rahalan) [reply constructs/Microsoft.Management/managementGroup-structure/deploy.json:194]: No. That is correct

File: arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep [hunk 12]
```diff
@@ -35,7 +35,7 @@ resource privateEndpoint_name 'Microsoft.Network/privateEndpoints@2021-05-01' =
     customDnsConfigs: privateEndpoint_var.customDnsConfigs
   }
 
-  resource privateEndpoint_name_default 'privateDnsZoneGroups@2020-05-01' = if (!empty(privateEndpoint_var.privateDnsZoneResourceIds)) {
+  resource privateEndpoint_name_default 'privateDnsZoneGroups@2017-04-01' = if (!empty(privateEndpoint_var.privateDnsZoneResourceIds)) {
```
Comment (MrMCake) [inline arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep:38]: Was that intended?

File: arm/Microsoft.Network/expressRouteCircuits/deploy.bicep [hunk 13]
```diff
@@ -205,4 +205,4 @@ output expressRouteCircuitResourceGroup string = resourceGroup().name
 output expressRouteCircuitName string = expressRouteCircuits.name
 
 @description('The service key of the express route circuit')
-output expressRouteCircuitServiceKey string = reference(expressRouteCircuits.id, '2020-05-01').serviceKey
+output expressRouteCircuitServiceKey string = reference(expressRouteCircuits.id, '2017-04-01').serviceKey
```
Comment (MrMCake) [inline arm/Microsoft.Network/expressRouteCircuits/deploy.bicep:208]: Was that intended?

File: arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep [hunk 14]
```diff
@@ -35,7 +35,7 @@ resource privateEndpoint 'Microsoft.Network/privateEndpoints@2021-03-01' = {
     customDnsConfigs: privateEndpoint_var.customDnsConfigs
   }
 
-  resource privateDnsZoneGroups 'privateDnsZoneGroups@2020-05-01' = {
+  resource privateDnsZoneGroups 'privateDnsZoneGroups@2017-04-01' = {
```
Comment (MrMCake) [inline arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep:38]: Was that intended?

File: arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep (hunk 12 above)
Comment (rahalan) [reply arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep:38]: No. Will fix it

File: arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep (hunk 14 above)
Comment (rahalan) [reply arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep:38]: No. Will fix it

File: arm/Microsoft.Network/expressRouteCircuits/deploy.bicep (hunk 13 above)
Comment (rahalan) [reply arm/Microsoft.Network/expressRouteCircuits/deploy.bicep:208]: No. Will fix it

File: arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep (hunk 14 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep]: Please re-run the pipeline when upating this to a new API version to be sure it works

File: arm/Microsoft.Network/expressRouteCircuits/deploy.bicep (hunk 13 above)
Comment (MrMCake) [reply arm/Microsoft.Network/expressRouteCircuits/deploy.bicep]: Please re-run the pipeline when upating this to a new API version to be sure it works

File: arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep (hunk 14 above)
Comment (rahalan) [reply arm/Microsoft.Insights/privateLinkScopes/.bicep/nested_privateEndpoint.bicep]: [Insights: PrivateLinkScopes](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.privatelinkscopes.yml)

File: arm/Microsoft.Network/expressRouteCircuits/deploy.bicep (hunk 13 above)
Comment (rahalan) [reply arm/Microsoft.Network/expressRouteCircuits/deploy.bicep]: [Network: ExpressRouteCircuits](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.expressroutecircuits.yml)

File: arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep (hunk 12 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep]: Just saw you also updated this one to a newer one. Please also test

Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/.bicep/nested_privateEndpoints.bicep]: Then again - the parameter file doesn't test it anyways (though it should)

**Review by MrMCake — APPROVED** [review]

**Pull request merged**